### ISSUE TITLE
Fast tasks cause the task output terminal to automatically clear & close in tasks.json version 2.0.0

### ISSUE THREAD

**Issue by SE2Dev** [Microsoft/vscode#33858]
## System Information:

- VSCode Version: 1.15.1
- OS Version: Debian 9.1 (Stretch)

 - Reproduces without extensions: Yes

**Note:** I've attempted to reproduce this issue on Windows, but have been unable to do so, I'm not sure if it's because executing the command takes longer on Windows or if its because the issue simply isn't present on Windows.

## Steps to Reproduce:

1. Create a new workspace
1. Create a tasks.json with the following contents (note that it's version 2.0.0)
	```
	{
		// See https://go.microsoft.com/fwlink/?LinkId=733558
		// for the documentation about the tasks.json format
		"version": "2.0.0",
		"tasks": [
			{
				"taskName": "test",
				"type": "shell",
				"command": "echo hello world; sleep 1",
				"group": {
					"kind": "test",
					"isDefault": true
				}
			}
		]
	}
	```
1. Run the test test several times, note how the terminal remains visible after the task has completed, and contains the following message: 
![image](https://user-images.githubusercontent.com/5873790/30079779-550467f6-924f-11e7-9531-64892f51878b.png)
1. Remove the sleep portion of the command so that the command line of tasks.json reads as follows:
`"command": "echo hello world"`
1. Run the test task several times, note that if you still had the terminal open from the previous step, that it gets automatically closed after the task completes. The terminal re-opens every time the new version of the command is run, and is automatically closed afterwards. The following is all that is printed (note that the "Terminal will be reused by tasks, press any key to close it" message is not displayed)
![image](https://user-images.githubusercontent.com/5873790/30079964-0cba22b4-9250-11e7-931e-7f90ab6d62c3.png)

## Additional Info
This is actually a combination of two parts of the same issue, the first being that in tasks.json version 2.0.0 a task which terminates too quickly will automatically clear and close the tasks output terminal. In tasks.json version 0.1.0, the expected behavior (of the task output terminal remaining open) occurs regardless of whether or not the task completes extremely quickly.

I have also included a [zip archive](https://github.com/Microsoft/vscode/files/1278719/example.zip) containing two versions of an example workspaces where this problem is present; the workspaces contain the source code for a basic "Hello World" program, a makefile to build it, and the appropriate tasks.json. The `tasks-new` workspace uses tasks.json version 2.0.0 whereas 'tasks-legacy' uses tasks.json version 0.1.0. The behavior I've experienced in these workspaces is as follows:

| Workspace 	| Build Command (In a clean workspace) 	| Build Command (When the program has already been compiled) 	| Test Command 	| Test Command (When the example program has been built with the sleep(1) line enabled) 	|
|--------------	|-------------------------------------------------------------------------------------------------------------------------------------------	|------------------------------------------------------------------------------------------------------	|----------------------------------------------------------------------------------	|--------------------------------------------------------------------------------------------------------------------------------------------------	|
| tasks-legacy 	| Build output is displayed, the terminal remains open 	| "make: Nothing to be done for 'build'." is displayed, the terminal remains open 	| "[0] './bin/a.out'" is displayed, the terminal remains open 	| "[0] './bin/a.out'" is displayed, the terminal remains open 	|
| tasks-new 	| Build output is displayed, the terminal remains open, "Terminal will be reused by tasks, press any key to close it." message is displayed 	| "make: Nothing to be done for 'build'." is displayed, the terminal is closed when the task completes 	| "[0] './bin/a.out'" is displayed, the terminal is closed when the task completes 	| "[0] './bin/a.out'" is displayed, the terminal remains open, "Terminal will be reused by tasks, press any key to close it." message is displayed 	|
|  	|  	|  	|  	|  	|

## Files
[example.zip](https://github.com/Microsoft/vscode/files/1278719/example.zip)

**Comment by dbaeumer** [comment]
Thanks a lot for the details PR report. This is a duplicate of #32276 and is addressed in the upcoming 1.16 release

**Comment by SE2Dev** [comment]
Excellent - Thanks for the info! Sorry about making a duplicate, I didn't see the other issue when I checked.
